### PR TITLE
Add custom propTypes shape to Caption and Paragraph components

### LINKED ISSUE #425: Create prop types shape for typography scripts
**Is your feature request related to a problem? Please describe.**
We should be more restrictive about what 'object' we allow

Current:
```
SubHeading.propTypes = {
  script: objectOf(object).required,
};
```

**Describe the solution you'd like**
Following the [articles example](https://github.com/bbc/simorgh/blob/latest/src/app/containers/Article/index.jsx#L64), export a proptypes shape to use in downstream components

Desired:

```
SubHeading.propTypes = {
  script: objectOf(scriptPropTypes).required,
};
```

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: When run locally, you dont get a prop type warning.

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #425

**Overall change:** 
We should be more restrictive about what script 'object' we allow to pass to the component.
For this reason, we should add a custom propType shape matching the current scripts values.

**Code changes:**
- Replace `defaultProps` with `propTypes` in the `Caption` and `Paragraph` components
- Import script `propType` shape from `gel-foundations`

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-caption/src/index.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 } from '@bbc/gel-foundations/typography';
```
Comment (dr3) [inline packages/components/psammead-caption/src/index.jsx]: Not sure why theres a blank space here
```suggestion
```

File: packages/components/psammead-caption/src/index.stories.jsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ storiesOf('Caption', module)
     inputProvider(
       ['visually hidden text', 'caption'],
       (hiddenText, captionText) => (
-        <Caption>
+        <Caption script={latin}>
```
Comment (dr3) [inline packages/components/psammead-caption/src/index.stories.jsx:25]: Good spot :P

**Review by dr3 — COMMENTED** [review]
Just 1 blank line annoyance, LGTM if jenkins passes

**Review by dr3 — DISMISSED** [review]

**Review by sareh — CHANGES_REQUESTED** [review]
The changes so far look great. I noticed that the `psammead-headings` component also needs this update: https://github.com/bbc/psammead/blob/9e5fdea65f923d4827311ec9dc7171adbe13d069/packages/components/psammead-headings/src/index.jsx#L46-L52

Could you please also add this to the PR?

**Comment by DenisHdz** [comment]
> The changes so far look great. I noticed that the `psammead-headings` component also needs this update:
> 
> [psammead/packages/components/psammead-headings/src/index.jsx](https://github.com/bbc/psammead/blob/9e5fdea65f923d4827311ec9dc7171adbe13d069/packages/components/psammead-headings/src/index.jsx#L46-L52)
> 
> Lines 46 to 52 in [9e5fdea](/bbc/psammead/commit/9e5fdea65f923d4827311ec9dc7171adbe13d069)
> 
>  Headline.defaultProps = { 
>    script: objectOf(object).required, 
>  }; 
>   
>  SubHeading.defaultProps = { 
>    script: objectOf(object).required, 
>  }; 
> Could you please also add this to the PR?

@sareh There is already a PR in progress for the Headings component https://github.com/bbc/psammead/pull/428

**Review by ghost — DISMISSED** [review]
👍

File: packages/components/psammead-caption/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-caption",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (ghost) [inline packages/components/psammead-caption/package.json]: Update `package.json` . #446 has been merged 
```suggestion
  "version": "1.1.1",
```

File: packages/components/psammead-caption/package-lock.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-caption",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (ghost) [inline packages/components/psammead-caption/package-lock.json]: Update
```suggestion
  "version": "1.1.1",
```

File: packages/components/psammead-caption/CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
-<<<<<<< HEAD
+| 1.0.2   | [PR#459](https://github.com/bbc/psammead/pull/459) Add custom propTypes shape |
```
Comment (ghost) [inline packages/components/psammead-caption/CHANGELOG.md]: Update `CHANGELOG.md`. #446 has been merged.

**Review by ghost — COMMENTED** [review]
Just a few changes to resolve possible merge conflict.

**Review by ghost — APPROVED** [review]
Looks great.

**Review by dr3 — APPROVED** [review]

**Review by sareh — APPROVED** [review]

**Comment by david-boydell** [comment]
LGTM

**Pull request merged**